### PR TITLE
Changes for eval2

### PR DESCRIPTION
**Pull request by Hyperion101010**
Seems like the bugs are now taken care of, I would like to merge my changes in the official repo.

### CODE REVIEW

File: .github/workflows/rust.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - run: cargo build
+        working-directory: solang-server
+      - run: cargo fmt --all
```
Comment (seanyoung) [inline .github/workflows/rust.yaml:21]: I think it's a good idea to have "cargo clippy" here as well.

Here you have cargo build *before* cargo fmt, which is a good idea. I think I'll steal that for the solang repo!

File: solang-server/src/ServerUtils/backend.rs [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+#[derive(Debug, Default)]
+pub struct Backend {
+    state: Vec<usize>,
+}
+
+#[warn(dead_code)]
+impl Backend {
+    pub fn search_line(data: &str, val: usize) -> (usize, usize) {
+        let mut line = Vec::new();
```
Comment (seanyoung) [inline solang-server/src/ServerUtils/backend.rs]: - This function does not need to be `pub`. 
- The name is not descriptive. How about `file_offset_to_line_column` (just a suggestion)
- Function needs doccomment explaining what it does
- The idea of building of building the `line` vector and then doing binary search is that it is faster when doing multiple scans. Here you build the `line` vector for each call, which is actually less efficient than a straight scan of the file.

I think it should either cache the `line` vector somewhere or just scan the data until offset for the right number of lines.

File: solang-server/src/ServerUtils/backend.rs [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+                l = mid + 1;
+            } else {
+                r = mid - 1;
+            }
+        }
+        let col = val - line[count - 1];
+        (count, col - 1)
+    }
+
+    pub fn convert_diagnos(ns: ast::Namespace, filecache: &mut FileCache) -> Vec<Diagnostic> {
```
Comment (seanyoung) [inline solang-server/src/ServerUtils/backend.rs]: - Not sure this function needs to be public.
- Abbreviating diagnositic to diagnos is a bit confusing
- Needs a doccomment explaining what it does

Every function in this file could do with doccomment explaining what it does

File: solang-server/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+use tower_lsp::{LspService, Server};
+
+#[allow(non_snake_case)]
+mod ServerUtils;
```
Comment (seanyoung) [inline solang-server/src/main.rs]: Can we rename `ServerUtils` into something rust accepts without `#[allow(non_snake_case)]` please

**Review by seanyoung — CHANGES_REQUESTED** [review]
Looks great, some minor changes please.

Now that `tower-lsp` directory is no longer used, please also remove it.

File: .github/workflows/rust.yaml (hunk 1 above)
Comment (Hyperion101010) [reply .github/workflows/rust.yaml:21]: Sure, ha ha alright!

Comment (Hyperion101010) [reply .github/workflows/rust.yaml:21]: done.

File: solang-server/src/ServerUtils/backend.rs (hunk 2 above)
Comment (Hyperion101010) [reply solang-server/src/ServerUtils/backend.rs]: done.

File: solang-server/src/main.rs (hunk 4 above)
Comment (Hyperion101010) [reply solang-server/src/main.rs]: done

File: solang-server/src/ServerUtils/backend.rs (hunk 3 above)
Comment (Hyperion101010) [reply solang-server/src/ServerUtils/backend.rs]: done

File: solang-server/src/serverutils/backend.rs [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+
+            let range = Range::new(p1, p2);
+
+            let diagnostic = &diag;
+
+            let sev = match diagnostic.level {
+                ast::Level::Info => DiagnosticSeverity::Information,
+                ast::Level::Warning => DiagnosticSeverity::Warning,
+                ast::Level::Error => DiagnosticSeverity::Error,
+                ast::Level::Debug => DiagnosticSeverity::Hint,
```
Comment (seanyoung) [inline solang-server/src/serverutils/backend.rs]: debug is not a hint, debug diagnostics should be ignored.

**Review by seanyoung — CHANGES_REQUESTED** [review]
tower-lsp is still present
PR cannot be merged (see github message)

File: solang-server/src/serverutils/backend.rs (hunk 5 above)
Comment (Hyperion101010) [reply solang-server/src/serverutils/backend.rs]: done

**Review by seanyoung — APPROVED** [review]

**Pull request merged**